server/middleware/appender.go

```diff
@@ -0,0 +1,109 @@
+package middleware
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/src-d/code-annotation/server/serializer"
+
+	"github.com/src-d/code-annotation/server/service"
+)
+
+func NewMiddlewaresWithAppender(ms []func(http.Handler) http.Handler) []func(http.Handler) http.Handler {
+	return append(ms, Appender)
+}
+
+func Appender(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		p := &postponed{
+			ResponseWriter: w,
+		}
+		defer p.Close()
+
+		next.ServeHTTP(p, r)
+
+		p.flush()
+	})
+}
+
+type postponed struct {
```

Comment:
wrapping `http.ResponseWriter` isn't such an easy task. You can read about it more [here](https://github.com/felixge/httpsnoop#why-this-package-exists) for example.

Reply:
I tried to emulate the same flow that [chi/middleware/compress](https://github.com/go-chi/chi/blob/master/middleware/compress.go) and [chi/middleware/logger](https://github.com/go-chi/chi/blob/master/middleware/logger.go) follows. Both wrap it to ensure the response is created in a controlled and homogeneous way.

Thanks for the resource, I'll read a bit more to consider unhandled circumstances.

Reply:
functionality dropped